code/modules/mob/living/silicon/robot/robot_damage.dm

```diff
@@ -153,3 +153,19 @@
 		burn	-= (picked.electronics_damage - burn_was)
 
 		parts -= picked
+
+/mob/living/silicon/robot/emp_act(severity)
+	if(components["surge"].installed)

```

Comment:
I am pretty sure that this is a dangerous access. So you'd be better off doing `if (components["surge"] && components["surge"].installed)`